client/allocrunner/taskrunner/service_hook.go

```diff
@@ -91,12 +95,17 @@ func (h *serviceHook) Poststart(ctx context.Context, req *interfaces.TaskPoststa
 	// Create task services struct with request's driver metadata
 	workloadServices := h.getWorkloadServices()
 
+	defer func() { h.initialRegistration = true }()
 	return h.consul.RegisterWorkload(workloadServices)
 }
 
 func (h *serviceHook) Update(ctx context.Context, req *interfaces.TaskUpdateRequest, _ *interfaces.TaskUpdateResponse) error {
 	h.mu.Lock()
 	defer h.mu.Unlock()
+	if !h.initialRegistration {
+		// no op since initial registration has not finished
+		return nil
+	}
```

Comment:
Might make sense to update service hook fields but avoid calling out to consul.  I'm thinking of the case where a client receives an in-place update of a task before it starts - we should honor the new `TaskEnv`/`ShutdownDelay`/etc in the PostStart invocation.